### PR TITLE
v8 add filterArea back

### PR DESCRIPTION
**Pull request by GoodBoyDigital**
also fix up alpha filter

### Summary
🎨📝🧪

This pull request updates the `AlphaFilter` class and the `FilterEffect` class to support the new filter system for WebGPU and WebGL. It also adds the `filterArea` property to both classes and the `EffectsMixin` interface, which allows specifying the area of the container to apply the filters to. It fixes some syntax and naming issues, and adds new test scenes for the `AlphaFilter` and the `filterArea` property.

> _Sing, O Muse, of the glorious deeds of the code warriors,_
> _Who with skill and wisdom crafted the filters of light and color,_
> _And adorned their containers with `filterArea` and `AlphaFilter`,_
> _Like Hephaestus who forged the shield of Achilles with divine artistry._

### Walkthrough
*  Add the AlphaFilter class for WebGL and WebGPU renderers, which applies a uniform alpha value to the input texture ([link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-cfd5d723ec1386f61fbf6f9dffe86403a5bf849062c677199c8ec04f4d5f66bdL1-R6), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-cfd5d723ec1386f61fbf6f9dffe86403a5bf849062c677199c8ec04f4d5f66bdL49-R58), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-cfd5d723ec1386f61fbf6f9dffe86403a5bf849062c677199c8ec04f4d5f66bdL55-R66), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-cfd5d723ec1386f61fbf6f9dffe86403a5bf849062c677199c8ec04f4d5f66bdL65-R76), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-bf7b3753c9903b1538f41b4b50c19f5a7de6fc78a9cc4cf50b342c65344cbc89R1-R12), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-caf7b0ccf33dcbac360f8447a73d558203709fbb540d3219c593865e8c992203L1-L6), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-caf7b0ccf33dcbac360f8447a73d558203709fbb540d3219c593865e8c992203L12-R7), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-caf7b0ccf33dcbac360f8447a73d558203709fbb540d3219c593865e8c992203L20-R32), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-caf7b0ccf33dcbac360f8447a73d558203709fbb540d3219c593865e8c992203L45-L49), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-caf7b0ccf33dcbac360f8447a73d558203709fbb540d3219c593865e8c992203L66-R56), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-caf7b0ccf33dcbac360f8447a73d558203709fbb540d3219c593865e8c992203L73-R67))
* Add the filterArea property to the FilterEffect class and the EffectsMixin interface, which allows specifying the area of the input texture to apply the filters to ([link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-da799ffa229b71851639ef368adc4e665ebb9e2fe0685e9a8ee670b72768875aR1), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-da799ffa229b71851639ef368adc4e665ebb9e2fe0685e9a8ee670b72768875aL7-R16), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-da799ffa229b71851639ef368adc4e665ebb9e2fe0685e9a8ee670b72768875aR27), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-da799ffa229b71851639ef368adc4e665ebb9e2fe0685e9a8ee670b72768875aL28-R38), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-8e6c7bee7983c3ce29fe2c1ba21299f6f658c65b05d3cf425dbb00c9e79a72b7L139-R156), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-f606b08d9454f6099cb00355e2c66d520e23ef316e92fdf94cfb7e4f2b4e888fR6), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-f606b08d9454f6099cb00355e2c66d520e23ef316e92fdf94cfb7e4f2b4e888fL18-R24), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-f606b08d9454f6099cb00355e2c66d520e23ef316e92fdf94cfb7e4f2b4e888fL98-R104), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-f606b08d9454f6099cb00355e2c66d520e23ef316e92fdf94cfb7e4f2b4e888fL113-R119), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-f606b08d9454f6099cb00355e2c66d520e23ef316e92fdf94cfb7e4f2b4e888fL123-R142))
* Add test scenes for the AlphaFilter class and the filterArea property, which check the visual output and the filter bounds ([link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-32d852678760110e04cef95784a931a028df627beb6c4b57ccc0815acc086c1dR1-R29), [link](https://github.com/pixijs/pixijs/pull/9781/files?diff=unified&w=0#diff-f64ca718f07319122b22b9d1003ae929c4f88eaff5d7f9aed2aba24424915b0cR1-R26))

### CODE REVIEW

File: tests/visual/scenes/filters/alpha-filter.scene.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { AlphaFilter } from '../../../../src/filters/defaults/alpha/AlphaFilter';
+import { Container } from '../../../../src/scene/container/Container';
+import { Graphics } from '../../../../src/scene/graphics/shared/Graphics';
+
+import type { TestScene } from '../../types';
+
+export const scene: TestScene = {
```
Comment (achamas-playco) [inline tests/visual/scenes/filters/alpha-filter.scene.ts:7]: Should we also have a png snapshot accompanying the test?

Comment (GoodBoyDigital) [reply tests/visual/scenes/filters/alpha-filter.scene.ts:7]: yup, uploaded 👍

File: tests/visual/scenes/filters/alpha-filter.scene.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    it: 'Alpha filter should be applied correctly',
+    only: true,
```
Comment (Zyie) [inline tests/visual/scenes/filters/alpha-filter.scene.ts:9]:
```suggestion
```

File: tests/visual/scenes/filters/filter-area.scene.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { ColorMatrixFilter } from '../../../../src/filters/defaults/color-matrix/ColorMatrixFilter';
+import { Rectangle } from '../../../../src/maths/shapes/Rectangle';
+import { Graphics } from '../../../../src/scene/graphics/shared/Graphics';
+
+import type { Container } from '../../../../src/scene/container/Container';
+import type { TestScene } from '../../types';
+
+export const scene: TestScene = {
+    it: 'Filter Area should be applied correctly',
+    only: true,
```
Comment (Zyie) [inline tests/visual/scenes/filters/filter-area.scene.ts:10]:
```suggestion
```

File: tests/visual/scenes/filters/alpha-filter.scene.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    create: async (scene: Container) =>
+    {
+        const filterContainer = new Container();
+
+        const rect = new Graphics().rect(0, 0, 100, 100).fill('orange');
+
+        const rect2 = new Graphics().rect(50, 50, 100, 100).fill('orange');
+
+        filterContainer.addChild(rect, rect2);
+        // filterContainer.alpha = 0.5;
```
Comment (Zyie) [inline tests/visual/scenes/filters/alpha-filter.scene.ts:18]:
```suggestion
```

**Pull request merged**